### PR TITLE
More work on Headphones integration

### PR DESCRIPTION
**Pull request by smcpeck**
This PR is to further address #1184.  It is ongoing for now, I only opened it so I could get a build with my changes and deploy/test them locally.

### CODE REVIEW

**Comment by smcpeck** [comment]
@tidusjar can you look at this and see what the heck I did? I can't even login anymore and it makes little sense to me. Very tough to debug.

**Comment by tidusjar** [comment]
Cannot log into Ombi?

**Comment by smcpeck** [comment]
Correct.
When logging in, I see the `/userlogin/password` endpoint come back with a 500 and...
```"Nancy.RequestExecutionException: Oh noes! ---> System.NullReferenceException: Object reference not set to an instance of an object
  at Ombi.UI.Modules.UserLoginModule+<PasswordLogin>d__44.MoveNext () [0x00490] in <5ceb56282d8a4b419459163a7e6dc95f>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0004e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x0002e] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x0000b] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () [0x00000] in <f712f98eb8e445c8918edaf595bbe465>:0 
  at Ombi.UI.Modules.UserLoginModule+<<-ctor>b__0_4>d.MoveNext () [0x00060] in <5ceb56282d8a4b419459163a7e6dc95f>:0 
   --- End of inner exception stack trace ---
  at Nancy.NancyEngine.InvokeOnErrorHook (Nancy.NancyContext context, Nancy.ErrorPipeline pipeline, System.Exception ex) [0x000bd] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 "```
Seems I've jammed up logging in somehow.

**Comment by tidusjar** [comment]
Looking at the files changed you have not made any code changes that would cause this. Have you done anything strange with your DB?

The stack `Ombi.UI.Modules.UserLoginModule+<PasswordLogin>d__44.MoveNext ()` implies it's failing in an iterator in the PasswordLogin method... 

Place a breakpoint on the entry and follow it through?

**Comment by smcpeck** [comment]
Current EAP build works fine against my database, so I think we can rule that out.  I'll look at live debugging in `PasswordLogin()` tonight.  Been trying to debug with extra logging, but it's not fun. Heh.

The only other recent change is that I've upgraded to `mono 4.8`...

Thanks for chiming in.

**Comment by smcpeck** [comment]
No idea what my login problem was -- all is well now.

Latest commits should fully fix #1184.

The only thing I don't like here is that extra non-HP stuff kinda got lumped into everything since I changed a bit in ApiRequest to help me log & debug better.

Let me know what you think.  This is ready to merge in my eyes.

File: Ombi.Core/HeadphonesSender.cs [hunk 1]
```diff
@@ -88,87 +88,20 @@ public HeadphonesSender(IHeadphonesApi api, HeadphonesSettings settings, IReques
         {
             var index = await Api.GetIndex(Settings.ApiKey, Settings.FullUri);
             var artistExists = index.Any(x => x.ArtistID == request.ArtistId);
+            bool artistAdd = false;
             if (!artistExists)
             {
-                var artistAdd = Api.AddArtist(Settings.ApiKey, Settings.FullUri, request.ArtistId);
```
Comment (smcpeck) [inline Ombi.Core/HeadphonesSender.cs]: @tidusjar 

Just to comment here...  This was the main defect associated much of the HP integration issues.  `artistAdd` was getting assigned a `Task<bool>` and things didn't really fall into place until that task was complete -- all of the extra polling and retrying was just unnecessary.

A convention we use at my day job is to **not** use `var` like this so frequently.  Variable types should always be explicitly obvious on the line of their declaration.  Had this code been explicit about `bool artistAdd` from the start, the code wouldn't have even compiled.

The only time I am a fan of `var` is when things are super obvious like:
```var thing = new Thing();```
or
```var something = GetObject<Something>(param1, param2);```

Just thought I'd share that as I think it helps code be more readable and prevent odd behaviors when you get unexpected types returns.

Cheers!

**Pull request merged**